src/main/assets/js/courts-table-search.ts

```diff
@@ -0,0 +1,220 @@
+  private static courtsTableHeaderDesc = 'courts-table-header-desc';
+  private static courtsTableHeaderInactive = 'courts-table-header-inactive';
+  private static courtsFrontendUrl = '#courtsFrontendUrl';
+  private static tableCourtsNameId = '#tableCourtsName';
+  private static tableCourtsUpdatedId = '#tableCourtsUpdated';
+
+  /**
+   *
+   * Sets up the table view for the main page, on load the default configuration
+   * is used, otherwise it is updated on each search toggle or courts search
+   *
+   * @param filterName Either 'name' or 'date' depending on the type of toggle
+   * @param defaultValue set to false by default on initial load
+   */
+  static setUpTable(filterName: string, defaultValue = false): void {
+    defaultValue
+      ? CourtsTableSearch.setUpTableData(
+        (document.getElementById(this.courtsHiddenId) as HTMLInputElement).value,
+      $(this.searchCourtsFilterId).val() as string, // if the user clicks 'back' in the browser after view or editing
+      false, orderToggleState.ASC, orderToggleState.INACTIVE)
+      : CourtsTableSearch.setUpTableData((
+        document.getElementById(this.courtsHiddenId) as HTMLInputElement).value,
+        $(this.searchCourtsFilterId).val() as string,
+        $(`#main-content input[name=${this.toggleClosedCourtsDisplay}]`).prop('checked'),
+        $(this.courtsNameAscToggleId).val() as string,
+        $(this.courtsUpdatedAscToggleId).val() as string);
+
+    filterName == 'name' ?
+      CourtsTableSearch.switchTableClasses($(this.courtsNameAscToggleId),
+        $(this.tableCourtsNameId), $(this.tableCourtsUpdatedId)) :
+      CourtsTableSearch.switchTableClasses($(this.courtsUpdatedAscToggleId),
+        $(this.tableCourtsUpdatedId), $(this.tableCourtsNameId));
+  }
+
+  /**
+   *
+   * Retrieves the filtered results from the toggles and search, and sets the html
+   * of the tbody element of the table accordingly
+   *
+   * @param courts the list of public court information
+   * @param searchFilterValue the search value from the 'search courts or tribunals' input field
+   * @param includeClosedCourts whether or not to include closed courts in the result
+   * @param orderNameAscendingFilter whether to sort by name asc or desc
+   * @param orderUpdatedAscendingFilter whether to sort by last updated date asc or desc
+   * @private
+   */
+  private static setUpTableData(courts: string, searchFilterValue: string, includeClosedCourts: boolean,
+    orderNameAscendingFilter: string, orderUpdatedAscendingFilter: string): void {
+    const filteredCourts = CourtsTableSearch.filterCourts(courts, searchFilterValue, includeClosedCourts,
+      orderNameAscendingFilter, orderUpdatedAscendingFilter);
+    $(this.courtsResultsSection).html(CourtsTableSearch.getCourtsTableBody(filteredCourts));
+    searchFilterValue.length ? $(this.numberOfCourts).show().text(
+      'Showing ' + filteredCourts.length + ' results')
+      : $(this.numberOfCourts).hide();
```

Comment:
Looking at 'scenario 5' in the requirements, should this be shown when there are no search params entered?

Reply:
Good shout! Asked Ade and you are right! So i have changed